### PR TITLE
support setting a note on an opslevel_service

### LINKED ISSUE #467: As a user, I would like to configure the "notes" section of a service using the provider
While implementing the service module i noticed that we don't allow you to configure the "notes" section of a service.

Screenshot 2024-08-30 at 7 20 40 AM

This looks to be a separate mutation using `ServiceNoteUpdateInput`

# AC

- [ ] I can use the following terraform to configure the "notes" of a service
```tf
resource "opslevel_service" "this" {
  name  = var.name
  notes  = "Hello World Notes!"
}
```

### PR DESCRIPTION
**Pull request by davidbloss**
Resolves #467  (depends on [PR #462](https://github.com/OpsLevel/opslevel-go/pull/462)

### Problem

Setting the "Notes" on a Service is not supported

### Solution

Support setting "Notes" on a Service

### Checklist

- [x] I have run this code, and it appears to resolve the stated issue.
- [x] This PR does not reduce total test coverage
- [x] This PR has no user interface changes or has already received approval from product management to change the interface.
- [x] Make a [changie](https://github.com/OpsLevel/terraform-provider-opslevel/blob/main/CONTRIBUTING.md#changie-change-log-generation) entry that explains the customer facing outcome of this change

### CODE REVIEW

File: opslevel/resource_opslevel_service.go [hunk 1]
```diff
@@ -471,3 +487,22 @@ func (r *ServiceResource) Delete(ctx context.Context, req resource.DeleteRequest
...
+func updateServiceNote(client opslevel.Client, service opslevel.Service, planModel ServiceResourceModel) (*opslevel.Service, error) {
+	if planModel.Note.ValueString() == service.Note {
+		return nil, nil
+	}
+
+	serviceNoteUpdateInput := opslevel.ServiceNoteUpdateInput{
+		Service: opslevel.IdentifierInput{Id: opslevel.RefOf(service.Id)},
+	}
+
+	if service.Note != "" && planModel.Note.IsNull() {
```
Comment (davidbloss) [inline opslevel/resource_opslevel_service.go:500]: Needs updating...

File: opslevel/resource_opslevel_service.go [hunk 2]
```diff
@@ -256,6 +262,13 @@ func (r *ServiceResource) Create(ctx context.Context, req resource.CreateRequest
 		}
 	}
 
+	if !planModel.Note.IsNull() {
```
Comment (rocktavious) [inline opslevel/resource_opslevel_service.go:265]: is this really needed during create?

File: .changes/unreleased/Feature-20240903-160040.yaml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+kind: Feature
+body: support setting a note on an opslevel_service
```
Comment (rocktavious) [inline .changes/unreleased/Feature-20240903-160040.yaml:2]:
```suggestion
body: "Adds support for setting the 'note' field on an 'opslevel_service' resource"
```

**Review by rocktavious — APPROVED** [review]

File: opslevel/resource_opslevel_service.go (hunk 2 above)
Comment (davidbloss) [reply opslevel/resource_opslevel_service.go:265]: Nope 😁 - I'll drop it

**Pull request merged**